test/executor.test.js

```diff
@@ -362,6 +362,14 @@ describe('executor', function() {
       }));
     }
 
+    it('should apply env passed in option object', function() {
+      function assertEnv(env) {
+        boot.execute(app, someInstructions({ env: env }));
+        assert.equal(app.get('env'), env);
+      }
+      assertEnv('test');
+    });
```

Comment:
What is the benefit of wrapping the test code in a function? I think the shorter version is better.

Also we are slowly abandoning `assert` in favour of `expect` from chai, could you please use it in all new code, even if the old code around uses `assert`?

Last but not least, you should use a variable that is not set in `NODE_ENV`. At the moment, your test should be always passing on our CI, providing my understanding is correct.

```js
it('should apply env passed in option object', function() {
  boot.execute(app, someInstructions({ env: 'custom-env' }));
  expect(app.get('env')).to.equal('custom-env');
});
```